test/spicepods/tpch/sf1/federated/databricks[sql_warehouse].yaml

```diff
@@ -0,0 +1,34 @@
+version: v1
+kind: Spicepod
+name: databricks[sql_warehouse]-federated
+
+datasets:
+  - from: databricks:spiceai_sandbox.tpch.partsupp
```

Comment:
nit: can we just define a `tpch` catalog instead?

Reply:
I think that changes the fully qualified table name and I don't know how the testoperator runs well enough to know if that is going to mess something up. I just followed the convention of the other databricks modes (e.g. spark and delta_lake)